haystack_experimental/components/tools/openapi/types.py

```diff
@@ -37,9 +38,41 @@ def path_to_operation_id(path: str, http_method: str = "get") -> str:
     :param http_method: The HTTP method to use for the operationId.
     :returns: The operationId.
     """
-    if http_method.lower() not in VALID_HTTP_METHODS:
-        raise ValueError(f"Invalid HTTP method: {http_method}")
-    return path.replace("/", "_").lstrip("_").rstrip("_") + "_" + http_method.lower()
+    return sanitize_function_name(path + "_" + http_method.lower())
+
+
+def sanitize(data: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Sanitizes the given function calling definition by adjusting its properties to LLM requirements.
+
+    :param data: The function calling definition(s) to sanitize.
+    :return: A sanitized function calling definition.
+    """
+    if isinstance(data, dict):
+        sanitized_data: Dict[str, Any] = {}
+        for key, value in data.items():
+            if key == "name" and isinstance(value, str):
```

Comment:
Why is the type check needed here? What are the expected types of the value?

Reply:
Can we formally codify this in some manner? I feel like we have far too many `Any` type hints w.r.t the OpenAPI specific code, and having some kind of schema (or at least better type annotation) would help make this more readable.

Reply:
I understand the sentiment @shadeMe . We have nicely, as much as needed imho, typed OpenAPISpec itself. Here we are dealing with these function definitions specified by various LLM providers. They look like this:
- Cohere https://docs.cohere.com/docs/tool-use#step-1
- Anthropic https://docs.anthropic.com/en/docs/build-with-claude/tool-use#tool-use-examples
- OpenAI https://cookbook.openai.com/examples/how_to_call_functions_with_chat_models#basic-concepts

In other words they are just dictionaries. And although it would be nice to bring some order and typing perhaps it is not worth it given how they change and get expanded with new options. Maybe just providing references to these dictionary structures would help a reader quite a lot.